Rubberduck.Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs

```diff
@@ -180,15 +200,102 @@ private void CheckThatThereIsNoOtherSameNameDeclarationInScopeInReferencingMetho
+        {
+            var firstReference = target.References.FirstOrDefault();
+            if (firstReference == null)
+            {
+                return false;
+            }
+
+            if (firstReference.ParentScoping.Equals(target.ParentScopeDeclaration))
+            {
+                //The variable is already in the same scope and consequently the identifier already refers to the declaration there.
+                return false;
+            }
+
+            if (GetSameNameVariablesInProcedure(target).FirstOrDefault() != null)
+            {
+                return false;
+            }
+
+            if (target.QualifiedModuleName.Equals(firstReference.QualifiedModuleName))
+            {
+                //The variable is a module variable in the same module.
+                //Since there is no local declaration with the same name in the procedure,
+                //the identifier already refers to the declaration inside the method. 
+                return true;
+            }
+
+            return false;
+        }
+
+        private IEnumerable<Declaration> GetSameNameDeclarationsInModule(Declaration target)
         {
-            return new MoveCloserToUsageModel(target);
+            var firstReference = target.References.FirstOrDefault();
+            if (firstReference == null)
+            {
+                return null;
+            }
+
+            var sameNameDeclarationsInModule = _declarationFinderProvider.DeclarationFinder
+                .MatchName(target.IdentifierName)
+                .Where(decl => decl.QualifiedModuleName.Equals(firstReference.QualifiedModuleName));
+
+            return sameNameDeclarationsInModule;
         }
 
-        private void MoveCloserToUsage(Declaration target)
+        private IEnumerable<Declaration> GetSameNameVariablesInProcedure(Declaration target)
         {
-            var model = Model(target);
-            _refactoringAction.Refactor(model);
+            var firstReference = target.References.FirstOrDefault();
+            if (firstReference == null)
+            {
+                return null;
+            }
+
+            var sameNameVariablesInProcedure = GetSameNameDeclarationsInModule(target)
+                .ToList()
+                .Where(decl => decl.DeclarationType == DeclarationType.Variable
+                               && decl.ParentScopeDeclaration.Equals(firstReference.ParentScoping));
+
+            return sameNameVariablesInProcedure;
+        }
+
+        static private string GetDefaultDeclarationStatement(Declaration target)
+        {
+            if (target.ParentDeclaration is ModuleDeclaration || IsStatic(target))
+            {
+                return Tokens.Static;
+            }
+
+            return Tokens.Dim;
         }
+
+        //TODO: Add IsStatic member to VariableDeclaration - this is a copy from Rubberduck.Refractorings.AssignmentNotUsedInspection
```

Comment:
not sure that should be a member of VariableDeclaration. An extension method seems more appropriate.

Reply:
I just copied this from the Rubberduck.CodeAnalysis because I didn't really know where to place this (Rubberduck.Parsing, CodeAnalysis, Refactoring, what ever..). Using an extension method seems to make sense.